dashboard/app/models/user.rb

```diff
@@ -518,14 +518,14 @@ def unpassed_progression_level?(script_level, user_levels)
     end
   end
 
-  # Returns the most recent (via created_at) activity for the specified level.
-  # TODO(asher): Change this to use UserLevel rather than Activity.
+  # Returns the most recent (via updated_at) user_level for the specified
+  # level.
   def last_attempt(level)
-    Activity.where(user_id: self.id, level_id: level.id).order('id desc').first
+    UserLevel.where(user_id: self.id, level_id: level.id).
+      order('updated_at DESC').
+      first
   end
```

Comment:
Note that we might want to refine this implementation, recalling that `Activity` has no context of a `Script`, whereas `UserLevel` does have such context. We could therefore pass in an optional script, filtering by it (if present).